### ISSUE TITLE
Update Java requirement to Java 11

### ISSUE THREAD

**Issue by rschlussel** [prestodb/presto#14873]
This blocks https://github.com/prestodb/presto/issues/14549#issuecomment-656571077 which has a dependency that requires java 11 support.  Presumably more cases like this will come up in the future.  Also, we may want to use java 11 features.

I know at facebook, we needed to do a lot of testing and change some java configurations in order to be able to successfully run with Java 11, so I don't think we want to change it without advanced warning or some guideline.

@sujay-jain any comments to share

@zhenxiao @vkorukanti what do you think?

**Comment by beinan** [comment]
An open PR I post earlier. https://github.com/prestodb/presto/pull/14009  There are still quite a few unit test failures need fixing.  

We have been planning to canary 1~2 presto cluster on java 11 in twitter to collect some runtime data.  But unfortunately we don't have bandwidth in h1 2020.  

Our motivation is that java 11 would have a better g1gc performance.  

We also wanna try enabling Graal.  For this, we expect a performance improvement of around 5% , as we have observed on other services

**Comment by mbasmanova** [comment]
@beinan Beinan, FYI, @sujay-jain Sujay and @bhhari Bhavani are rolling out Java 11 and Graal at FB at this very moment.

**Comment by beinan** [comment]
@mbasmanova just some updates: we're rolling out Java 11 to ~400 workers at Twitter.  So far so good.

**Comment by junyi1313** [comment]
@beinan Hi, Beinan, any updates about Java 11? We also want to run PrestoDB on Java 11. Can we compile the code and run on Java 11 directly? Thanks.

**Comment by beinan** [comment]
@junyi1313 Yes,  my change had been merged last year, and we've running presto on java 11 for almost a year in twitter.   Everything looks good, and the performance and stability looks better.

**Comment by junyi1313** [comment]
@beinan That sounds great. We will try it.

Besides, do we have a JVM config file template (e.g. [jvm-config](https://trino.io/docs/current/installation/deployment.html#jvm-config)), and an IDE guide for Java 11 (e.g. [Running Trino in your IDE](https://github.com/trinodb/trino#overview))? That will be very useful for the developers, especially for JVM performance tuning. 

Thanks.

**Comment by beinan** [comment]
Hey @junyi1313 ,  sure,  there is a jvm config template for prestodb. https://prestodb.io/docs/current/installation/deployment.html#jvm-config

Running presto in IDE (IDEA intelliJ): https://github.com/prestodb/presto/blob/master/README.md#running-presto-in-your-ide

I also post the jvm config I'm using for large heap (>=100GB)
`-server
-Xms200G
-Xmx200G
-XX:+PreserveFramePointer
-XX:-UseBiasedLocking
-XX:+UnlockExperimentalVMOptions
-XX:G1HeapRegionSize=32M
-XX:+UseG1GC
-XX:+ExplicitGCInvokesConcurrent
-XX:+UseGCOverheadLimit
-XX:MaxMetaspaceSize=1G
-XX:ReservedCodeCacheSize=250M
-XX:+AggressiveOpts
-XX:+HeapDumpOnOutOfMemoryError
-XX:+ExitOnOutOfMemoryError
-XX:+PrintGCDetails
-XX:+PrintSafepointStatistics
-XX:PrintSafepointStatisticsCount=1
-XX:-UseAdaptiveSizePolicy
-XX:+UnlockDiagnosticVMOptions
-XX:InitiatingHeapOccupancyPercent=40
-XX:+UseStringDeduplication
-XX:StringDeduplicationAgeThreshold=6
-verbose:gc
-Xlog:gc:var/log/gc.log
-Djdk.nio.maxCachedBufferSize=0
-Djdk.attach.allowAttachSelf=true`

**Comment by junyi1313** [comment]
Thank you @beinan, that's really helpful.

**Comment by beinan** [comment]
@junyi1313 np, let us know if you're seeing any failure and performance downgrading on java 11.  Thanks!

I'm also having a working branch for java 14 and zgc, but there are still lots of fixing needed.

**Comment by junyi1313** [comment]
@beinan sure, we will upgrade to java 11 soon.

**Comment by mbasmanova** [comment]
CC: @amitkdutta 

@rschlussel Is there anything left to do here? Do we document required Java version ?

CC: @tdcmeehan @steveburnett @elharo

**Comment by steveburnett** [comment]
> CC: @amitkdutta
> 
> @rschlussel Is there anything left to do here? Do we document required Java version ?
> 
> CC: @tdcmeehan @steveburnett @elharo

Yes, the required Java version is documented in the README for Presto. 

https://github.com/prestodb/presto/blob/master/README.md#requirements

**Comment by elharo** [comment]
Don't confuse building and supporting for Java 8 with running on Java 8. Presto runs fine on Java 11.

**Comment by elharo** [comment]
You can compile it with Java 11.
You can run it on Java 11.

You can also compile and run it on Java 8, 17, 21, and pretty much any version from 8 on. Perhaps one day Oracle will finally remove sun.misc.Unsafe, and we might want to think about what to do about that now. However current JDKs from 8 on are not a problem. 

I still don't see any reason presented why we should not support Java 8.

**Comment by elharo** [comment]
That's a spotbugs issue, not a compiler or code issue. Might just need a newer version of spotbugs. I think 4.8.6.1 is current.

If the newer version still doesn't work with Java 11, we can remove it. It's a nice to have, not a requirement.
